### PR TITLE
[12.0][ADD] user see only maintenance team where he is join in

### PR DESCRIPTION
**Pull request by Borruso**
Add ir.rule which it will be able to see only maintenance team where user is join in

### CODE REVIEW

File: base_maintenance_group/security/maintenance_security.xml [hunk 1]
```diff
@@ -21,4 +21,15 @@
         <field name="users" eval="[(4, ref('base.group_system'))]"/>
     </record>
 
+    <record model="ir.rule" id="only_subscribe_maintenance_team">
```
Comment (astirpe) [inline base_maintenance_group/security/maintenance_security.xml]: Please place `id` attribute before `model`, as stated in https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#xml-files:


```suggestion
    <record id="only_subscribe_maintenance_team" model="ir.rule">
```

**Review by astirpe — CHANGES_REQUESTED** [review]
Thanks for this PR.
Only a minor request to adopt the conventions as per the OCA guidelines.

**Comment by jarroyomorales** [comment]
Hi @Borruso! I dont see any problem in being able to see other teams in the company. It may be interesting for some reason.

**Comment by Borruso** [comment]
Could I have to make rule as well for version 13.0 ??

**Comment by astirpe** [comment]
> Could I have to make rule as well for version 13.0 ??

Let's see if this PR is accepted a merged first. In case, after merged, you could create a new PR with the porting to V13.

**Comment by dalonsod** [comment]
> Hi @Borruso! I dont see any problem in being able to see other teams in the company. It may be interesting for some reason.

@Borruso maybe this rule could be added within a `<data noupdate="1">` section. And IMO you should create another rule to grant total access for the `Full Access` group.

File: base_maintenance_group/security/maintenance_security.xml [hunk 2]
```diff
@@ -21,4 +21,30 @@
...
+            <field name="perm_write" eval="True"/>
+            <field name="perm_unlink" eval="True"/>
+            <field name="perm_create" eval="True"/>
+        </record>
+
+        <record id="all_maintenance_team" model="ir.rule">
+            <field name="name">All Maintenance Team</field>
+            <field name="model_id" ref="maintenance.model_maintenance_team"/>
+            <field name="groups" eval="[(6, 0, [ref('maintenance.group_equipment_manager')])]"/>
+            <field name="domain_force">[(1, 'in', 1)]</field>
```
Comment (dalonsod) [inline base_maintenance_group/security/maintenance_security.xml]: This should not work:
```suggestion
            <field name="domain_force">[(1, '=', 1)]</field>
```

**Review by dalonsod — CHANGES_REQUESTED** [review]

**Review by dalonsod — APPROVED** [review]
Tested on runbot 👍 
Not blocking, I only suggest a cosmetic change: for better understanding of `maintenance_security.xml` file after these changes: the already existing records should be placed under a `<data noupdate="0">` section

**Review by marcelsavegnago — APPROVED** [review]
LGTM

**Review by etobella — APPROVED** [review]

**Comment by dalonsod** [comment]
> Not blocking, I only suggest a cosmetic change: for better understanding of `maintenance_security.xml` file after these changes: the already existing records should be placed under a `<data noupdate="0">` section

Maybe I've not explained this well. The suggested change should be keep `<data noupdate="1">` for the recently added rules and include `<data noupdate="0">` for the previous code (the `res.groups` records), in order to make the code more readable. @etobella do you agree?

**Comment by Borruso** [comment]
> > Not blocking, I only suggest a cosmetic change: for better understanding of `maintenance_security.xml` file after these changes: the already existing records should be placed under a `<data noupdate="0">` section
> 
> Maybe I've not explained this well. The suggested change should be keep `<data noupdate="1">` for the recently added rules and include `<data noupdate="0">` for the previous code (the `res.groups` records), in order to make the code more readable. @etobella do you agree?

Sorry, I didn't understand. Should I wait @etobella or not?

**Comment by dalonsod** [comment]
@etobella approved your recent changes, thus I'd wait for him

File: base_maintenance_group/security/maintenance_security.xml [hunk 3]
```diff
@@ -21,4 +21,30 @@
         <field name="users" eval="[(4, ref('base.group_system'))]"/>
     </record>
 
+    <data noupdate="0">
```
Comment (etobella) [inline base_maintenance_group/security/maintenance_security.xml]: Sorry, noupdate should be 1, as usual on record rules

**Review by etobella — CHANGES_REQUESTED** [review]
Follow @dalonsod comments, It is just cosmetic but easier to understand. Also, I saw the noupdate and it should be 1, not 0 on the rule, as we could make a change there. If you want, you could create the data on a migration script in order to be sure that are created, but make them unactive on the creation as we don't want to break the old logic.

**Review by dalonsod — APPROVED** [review]
:+1:

**Comment by etobella** [comment]
If you update the module, you will not get the new rules on your system, maybe you need a migration script in order to add them

**Comment by etobella** [comment]
Here you can check an example:
https://github.com/OCA/community-data-files/blob/12.0/account_payment_unece/migrations/12.0.1.0.0/post-migration.py#L7

**Comment by etobella** [comment]
@Borruso Have you checked?

**Comment by Borruso** [comment]
> @Borruso Have you checked?

@etobella I checked and I understand that even though the rules are in noupdate, no migration script is needed since these rules do not exist in the system, the problem shows up when the rules already existed in the system and I changed, for example, name or domain

**Review by etobella — APPROVED** [review]
You are right! Tested locally, when upgrading, the rules are created

**Review by marcelsavegnago — APPROVED** [review]

**Comment by Borruso** [comment]
@astirpe could you check if your change request is ok?

**Review by astirpe — APPROVED** [review]
Code review ok. Not tested.

**Pull request merged**